pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -72,7 +72,12 @@ func NewDeployer(cfg types.Config, labels map[string]string) *Deployer {
 // outputs them to the applyDir, and runs `kpt live apply` against applyDir to create resources in the cluster.
 // `kpt live apply` supports automated pruning declaratively via resources in the applyDir.
 func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []build.Artifact) ([]string, error) {
-	manifests, err := k.renderManifests(ctx, out, builds)
+	flags, err := k.getKptFnRunArgs()
```

Comment:
do these change during the course of the run? could they be computed inside `renderManifests()` rather than passed in as a parameter?

or even better, could they just be stored on the deployer struct itself, since these methods also operate directly on the struct?

Reply:
These flags should not be placed inside the renderManifest. renderManifest is called by multiple functions (render, deploy). but the getKptFnRunArgs includes some logic that is different per command (mainly about the --dry-run).

Besides flags, the kpt config should also be validated before each function call. Right now, the validation only happens when the config is used.
that's less efficient and potentially could make the workspace dirty (e.g. the .tmp-dir is not cleaned up if something goes wrong).

My next PR will try refactoring the code and fix these issues.